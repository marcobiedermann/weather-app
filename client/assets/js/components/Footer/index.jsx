import PropTypes from 'prop-types';
import React from 'react';

const Footer = (props) => {
  const { children } = props;

  return (
    <div>
      {children}
    </div>
  );
};

Footer.propTypes = {
  children: PropTypes.node,
};

Footer.defaultProps = {
  children: null,
};

export default Footer;
