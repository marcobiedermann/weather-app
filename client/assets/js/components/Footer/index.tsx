import React, { FC } from 'react';

const Footer: FC = (props) => {
  return <div {...props} />;
};

export default Footer;
