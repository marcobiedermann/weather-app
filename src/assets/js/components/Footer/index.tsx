import React, { FC } from 'react';

const Footer: FC = (props) => <div {...props} />;

export default Footer;
