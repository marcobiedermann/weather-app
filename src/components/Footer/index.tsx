import React, { ReactNode } from 'react';

interface FooterProps {
  children: ReactNode;
}

function Footer(props: FooterProps): JSX.Element {
  return <div {...props} />;
}

export type { FooterProps };
export default Footer;
