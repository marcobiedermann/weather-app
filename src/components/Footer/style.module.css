.footer {
  font-size: 0.875rem;
  padding: 0.5em 3em 3em;
}

.footer > :first-child {
  margin-block-start: 0;
}

.footer > :last-child {
  margin-block-end: 0;
}