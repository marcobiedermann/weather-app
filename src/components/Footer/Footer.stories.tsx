import type { Meta, StoryObj } from '@storybook/react';
import Footer from '.';

const meta: Meta<typeof Footer> = {
  args: {
    children: 'Footer',
  },
  component: Footer,
};

type Story = StoryObj<typeof Footer>;

const Default: Story = {};

export { Default };
export default meta;
