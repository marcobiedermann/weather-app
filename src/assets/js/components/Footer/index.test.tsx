import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '.';

describe('components/Footer', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Footer>Content</Footer>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
