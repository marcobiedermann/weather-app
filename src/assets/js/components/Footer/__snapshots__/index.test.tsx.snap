// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Footer renders correctly 1`] = `
<DocumentFragment>
  <div>
    Content
  </div>
</DocumentFragment>
`;
